kafka-streams-partitioners/weighted-group-partitioner/src/main/java/org/hypertrace/core/kafkastreams/framework/partitioner/PartitionerConfigServiceCachingClient.java

```diff
@@ -0,0 +1,53 @@
+package org.hypertrace.core.kafkastreams.framework.partitioner;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.typesafe.config.Config;
+import io.grpc.ManagedChannel;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.grpcutils.client.GrpcChannelRegistry;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.core.grpcutils.context.RequestContextConstants;
+import org.hypertrace.partitioner.config.service.v1.GetPartitionerProfileRequest;
+import org.hypertrace.partitioner.config.service.v1.PartitionerConfigServiceGrpc;
+import org.hypertrace.partitioner.config.service.v1.PartitionerConfigServiceGrpc.PartitionerConfigServiceBlockingStub;
+
+@Slf4j
+public class PartitionerConfigServiceCachingClient implements PartitionerConfigServiceClient {
+  private final PartitionerConfigServiceBlockingStub service;
+  private final LoadingCache<String, WeightedGroupProfile> partitionerConfigProfileCache;
+
+  public PartitionerConfigServiceCachingClient(Config config, GrpcChannelRegistry registry) {
+    PartitionerConfigServiceClientConfig clientConfig =
+        PartitionerConfigServiceClientConfig.from(config);
+    ManagedChannel managedChannel =
+        registry.forPlaintextAddress(clientConfig.getHost(), clientConfig.getPort());
+    this.service = PartitionerConfigServiceGrpc.newBlockingStub(managedChannel);
```

Comment:
```suggestion
    Channel channel =
        registry.forPlaintextAddress(clientConfig.getHost(), clientConfig.getPort());
    this.service = PartitionerConfigServiceGrpc.newBlockingStub(channel);
```

Reply:
The ManagedChannel in this API is tech debt that we can't clean up without breaking it unfortunately, but it's an impl detail that should not be exposed.